realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -2413,12 +2423,14 @@ void BaseRealSenseNode::startMonitoring()
         std::mutex mu;
         std::unique_lock<std::mutex> lock(mu);
         while(_is_running) {
-            _cv.wait_for(lock, std::chrono::milliseconds(time_interval), [&]{return !_is_running;});
+            auto next_wakeup = std::chrono::steady_clock::now() + std::chrono::milliseconds(time_interval);
```

Comment:
As I understand, you changed the order to get more regular readouts? (to include the time of publishing itself?) Is this something we could try to upstream?